akka-http/src/main/scala/akka/http/scaladsl/server/directives/BasicDirectives.scala

```diff
@@ -398,6 +400,11 @@ trait BasicDirectives {
       ctx.request.entity.toStrict(timeout, maxBytes).flatMap { strictEntity =>
         val newCtx = ctx.mapRequest(_.copy(entity = strictEntity))
         inner(())(newCtx)
+      }.recover {
+        case _: TimeoutException =>
+          throw IllegalRequestException(
```

Comment:
Is `TimeoutException` specifically thrown when strictify-ing the request times out, or might it also occur when it's the generation of the response that times out? For the former indeed an 4xx Client Error `IllegalRequestException` is approprate, for the latter arguably a 5xx Server Error might be better.

Do we have a test for that scenario? If not that might be useful to add.

Reply:
That's a good point. The solution then would be to swap the flatMap and the recover so that the recover only covers the `toStrict` and not the handling further down.

Reply:
I will cover this